applications/FluidDynamicsApplication/custom_strategies/strategies/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h

```diff
@@ -326,8 +326,21 @@ namespace Kratos {
 
                         array_1d<double, 3 > & OldVelocity = (itNode)->FastGetSolutionStepValue(VELOCITY, 1);
 
-                        noalias(itNode->FastGetSolutionStepValue(MESH_VELOCITY)) = itNode->FastGetSolutionStepValue(VELOCITY);
-                        UpdateDisplacement(CurrentDisplacement, OldDisplacement, OldVelocity, OldAcceleration, CurrentAcceleration);
+    
+                        
+                        if((itNode)->FastGetSolutionStepValue(IS_LAGRANGIAN_INLET) == 0.0)
```

Comment:
@elafnm I just saw that you are comparing a double with `==` 
this is dangerous due to machine accuracy, it might or might not work
you should do sth like the following:
`if((itNode)->FastGetSolutionStepValue(IS_LAGRANGIAN_INLET) < 1e-15)`

Reply:
Thank you dear @philbucher  for this important observation,

I just changed it as recommended.

with our best regards,

Reply:
Very good.
I think I saw it also in some other files, so you might want to check those too

Reply:
@philbucher 

will do dear, and will keep it in mind for future as well.

with my best regards,